src/App.js

```diff
@@ -1,25 +1,31 @@
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+import { Route } from 'react-router-dom';
+import countries from './countries';
+import Navbar from './components/Navbar';
+import CountriesList from './components/CountriesList';
+import CountryDetails from './components/CountryDetails';
 
-export default App;
+export default class App extends React.Component {
+  state = {
+    countries: []
+  }
+  componentDidMount() {
+    this.setState({
+      countries: countries
+    })
+  }
+  render() {
+    return (
+      <div className="App">
+        <Navbar />
+        <div className="container">
+          <div className="row">
+            <CountriesList countries={this.state.countries} />
+            <Route exact path="/:id" component={CountryDetails} />
```

Comment:
You could also pass the state here to the `CountryDetails` component using:
`<Route exact path="/:id" render={(props)=> <CountryDetails {...props} countries={this.state.countries}/>} />`